pandas/tests/strings/test_cat.py

```diff
@@ -376,3 +376,11 @@ def test_cat_different_classes(klass):
     result = s.str.cat(klass(["x", "y", "z"]))
     expected = Series(["ax", "by", "cz"])
     tm.assert_series_equal(result, expected)
+
+
+def test_cat_on_series_dot_str():
+    ps = Series(["AbC", "de", "FGHI", "j", "kLLLm"])
+
+    res = ps.str.cat(others=ps.str)
```

Comment:
Like the original issue mentioned https://github.com/pandas-dev/pandas/issues/28277#issuecomment-527851842=, `ps.str.cat(others=ps.str)` should raise an exception here. `ps.str.cat(others=ps)` should be okay

Reply:
Oh right, I was misled by the discussion in #29909. Should we fix this issue independently or wait until `Series.str.__iter__` was deprecated?

Reply:
Can you determine what the behavior when `Series.str.__iter__` will be removed? i.e. will a sensible error raise if `Series.str.__iter__` is removed?

Reply:
Right. `Series.str.__iter__` is already deprecated and will be removed in 2.0.

I'm asking _if_ you remove `Series.str.__iter__` locally and run the test case again, would a sensible error raise or would it make sense to do some input validation in `cat` to raise a better error message.

Reply:
Okay then in that case, could you just add a unit test in this PR? The unit test could exercise that the warning of `Series.str.__iter__` when `ps.str.cat(others=ps)` is run, but also have commented out code that would test the `TypeError` exception when `Series.str.__iter__` is removed

Reply:
No problem.